napari/_qt/layer_controls/qt_layer_controls_base.py

```diff
@@ -72,6 +80,71 @@ def __init__(self, layer):
         blend_comboBox.activated[str].connect(self.changeBlending)
         self.blendComboBox = blend_comboBox
 
+    def _populate_grid(self, *grid_scheme):
+        """
+        Populate the layer control grid layout based on a scheme in the form:
+        [
+            (Qwidget, ..., QWidget),
+            (...,),
+            (None, QWidget,)
+        ]
+
+        Ellipsis means "increase the column span of the previous widget" if part of a row, and
+        "increase the row span of all the widgets in the previous row" if part of a column.
+        Trailing Ellipses are assumed for rows with fewer entries than others.
+
+        None leaves an empty space in the grid.
+        """
+        # arguments that will be passed to grid.addWidget
+        # list of tuples in the form (widget, row, column, row_span, column_span)
+        args = []
+
+        max_cols = max(len(row) for row in grid_scheme)
+        previous_row = -1
+        for row, widgets in enumerate(grid_scheme):
+            if widgets == (None,):
+                continue
+            if widgets == (Ellipsis,):
+                # increment row span of all widgets in the previous row
+                for arg in args:
+                    if arg[1] == previous_row:
+                        arg[3] += 1
+                continue
+            previous_row = row
+
+            previous_col = -1
+            for col, wdg in enumerate(widgets):
+                if wdg is None:
+                    continue
+                if wdg is Ellipsis:
+                    # increment column span of the previous widget
+                    for arg in args:
+                        if arg[1] == row and arg[2] == previous_col:
+                            arg[4] += 1
+                    continue
+                previous_col = col
+
+                # generate simple label with translation if only a string was passed
+                if isinstance(wdg, str):
+                    wdg = QLabel(trans._(wdg))
+
```

Comment:
I would make sure that here wdg is a widget, we can imagine someone passing `(..., something_else)`, which would be invalid.

Reply:
Added a check.